File: src/main/java/com/sesame/domain/internal/ParameterType.java
package com.sesame.domain.internal;

import java.util.Arrays;

public enum ParameterType {

    BOOLEAN("boolean", boolean.class, true),
    CHAR("char", char.class, true),
    BYTE("byte", byte.class, true),
    SHORT("short", short.class, true),
    INT("int", int.class, true),
    LONG("long", long.class, true),
    FLOAT("float", float.class, true),
    DOUBLE("double", double.class, true),
    STRING("string", String.class, false),
    OBJECT("object", Object.class, false);

    private String displayType;
    private Class internalType;
    private boolean isPrimitive;

    ParameterType(String displayType, Class internalType, boolean isPrimitive) {
        this.displayType = displayType;
        this.internalType = internalType;
        this.isPrimitive = isPrimitive;
    }

    public String getDisplayType() {
        return this.displayType;
    }

    public Class getInternalType() {
        return this.internalType;
    }

    public boolean isPrimitive() {
        return isPrimitive;
    }

    public static ParameterType getInternalType(String displayType) {

        return Arrays.stream(ParameterType.values())
                .filter(parameterType -> parameterType.displayType.equalsIgnoreCase(displayType))
                .findFirst()
                .orElse(OBJECT);

    }
}

File: src/main/kotlin/com/sesame/core/domain/ParameterType.kt
package com.sesame.core.domain

import java.util.*

enum class ParameterType(
        val displayType: String,
        val internalType: Class<*>,
        val isPrimitive: Boolean) {

    BOOLEAN("boolean", Boolean::class.java, true),
    CHAR("char", Char::class.java, true),
    BYTE("byte", Byte::class.java, true),
    SHORT("short", Short::class.java, true),
    INT("int", Int::class.java, true),
    LONG("long", Long::class.java, true),
    FLOAT("float", Float::class.java, true),
    DOUBLE("double", Double::class.java, true),
    STRING("string", String::class.java, false),
    OBJECT("object", Object::class.java, false);

    companion object {

        fun getInternalType(displayType: String?): ParameterType {
            return Arrays.stream(values())
                    .filter { parameterType: ParameterType -> parameterType.displayType.equals(displayType, ignoreCase = true) }
                    .findFirst()
                    .orElse(OBJECT)
        }

    }

}